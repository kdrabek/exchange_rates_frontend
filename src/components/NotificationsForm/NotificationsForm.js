import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormControl, Button, FormGroup, ControlLabel, Form } from 'react-bootstrap';

class NotificationsForm extends Component {
  render() {
    const currencySelect = this.props.currencies.length ? this.props.currencies.map(currency => {
      return <option key={currency.code} value={currency.code}>{currency.code} ({currency.country})</option>
      }
    ) : <option></option>;
    return (
      <div>
        <Form onSubmit={this.props.submitForm}>
          <FormGroup>
            <ControlLabel>Waluta</ControlLabel>
            <FormControl componentClass="select" placeholder="select" name="currency">
              {currencySelect}
            </FormControl>
          </FormGroup>

          <FormGroup>
            <ControlLabel>Próg</ControlLabel>
            <FormControl type="number" step="0.01" min="0" placeholder="0.00" name="rate" required/>
          </FormGroup>

          <FormGroup>
            <ControlLabel>Warunek</ControlLabel>
            <FormControl componentClass="select" placeholder="select" name="threshold">
              <option value="ABOVE" selected>powyżej</option>
              <option value="BELOW">poniżej</option>
            </FormControl>
          </FormGroup>

          <FormGroup>
            <Button type="submit" bsStyle="success">Dodaj</Button>
          </FormGroup>
        </Form>       
      </div>
    );
  }
}

NotificationsForm.propTypes = {
  currencies: PropTypes.array,
  submitForm: PropTypes.func,
}

export default NotificationsForm;