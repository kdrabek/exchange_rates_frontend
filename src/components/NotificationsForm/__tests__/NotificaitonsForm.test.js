import React from 'react';
import ReactDOM from 'react-dom';
import NotificationsForm from '../NotificationsForm';
import renderer from 'react-test-renderer';

describe('Tagline component', () => {

  it('matches the snapshot', () => {
    const currencies = [
      {
        code: "ABC",
        name: "awesome currency",
        country: "country",
        table_type: "A"
      }, 
      {
        code: "CDE",
        name: "moar awesome currency",
        country: "totally different country",
        table_type: "A"
      }
    ];
    const component = renderer.create(
      <NotificationsForm currencies={currencies} submitForm={() => {}}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});