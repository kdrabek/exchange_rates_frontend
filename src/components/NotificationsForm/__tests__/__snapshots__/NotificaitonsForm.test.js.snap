exports[`NotificationsForm component matches the snapshot 1`] = `
<div>
  <form
    className=""
    onSubmit={[Function]}>
    <div
      className="form-group">
      <label
        className="control-label"
        htmlFor={undefined}>
        Waluta
      </label>
      <select
        className="form-control"
        id={undefined}
        name="currency"
        placeholder="select"
        type={undefined}>
        <option
          value="ABC">
          ABC
           (
          country
          )
        </option>
        <option
          value="CDE">
          CDE
           (
          totally different country
          )
        </option>
      </select>
    </div>
    <div
      className="form-group">
      <label
        className="control-label"
        htmlFor={undefined}>
        Próg
      </label>
      <input
        className="form-control"
        id={undefined}
        min="0"
        name="rate"
        placeholder="0.00"
        required={true}
        step="0.01"
        type="number" />
    </div>
    <div
      className="form-group">
      <label
        className="control-label"
        htmlFor={undefined}>
        Warunek
      </label>
      <select
        className="form-control"
        id={undefined}
        name="threshold"
        placeholder="select"
        type={undefined}>
        <option
          selected={true}
          value="ABOVE">
          powyżej
        </option>
        <option
          value="BELOW">
          poniżej
        </option>
      </select>
    </div>
    <div
      className="form-group">
      <button
        className="btn btn-success"
        disabled={false}
        type="submit">
        Dodaj
      </button>
    </div>
  </form>
</div>
`;
